fix(tests): align test contracts with new validation script rules

aliersh opened the pull request:
This PR updates several test files to pass the new validation rules introduced in the `test-validation` script.

### Summary of Changes

- **Contract Renaming**: 
  - Updated contract names to match expected validation patterns (e.g., `<Contract>_<Function>_Test`)
  - Renamed contracts such as `SystemConfig_Test` to `SystemConfig_SuperchainConfig_Test` and fixed pluralization issues like `SupportsInterfaces` → `SupportsInterface`

- **Function Name Alignment**: 
  - Renamed functions within test contracts to match their respective contract names (e.g., `test_supportsInterfaces_succeeds` → `test_supportsInterface_succeeds`)

- **Consolidation of Test Logic**:
  - Merged separate contracts into main test contracts where appropriate (e.g., consolidated L2 withdrawal tests into `SequencerFeeVault_Withdraw_Test`)
  - Introduced internal helper functions to reduce duplication

- **Validation Script Updates**:
  - Removed exclusions for test files that now follow validation rules
  - Updated comments for excluded files to reflect current validation status

aliersh committed:
fix(tests): align test contracts with new validation script rules

- Renamed test contracts to match expected naming patterns (e.g., <Contract>_<Function>_Test)
- Updated function names to align with contract names and validation expectations
- Merged standalone test contracts into primary test contracts to reduce fragmentation
- Refined test setup logic by introducing shared helper functions
- Removed test files from exclusion list in validation script now that they conform
- Adjusted comments in the validation script to reflect current file status

smartcontracts commented on packages/contracts-bedrock/test/L1/SystemConfig.t.sol:
```diff
@@ -734,9 +734,10 @@ contract SystemConfig_Guardian_Test is SystemConfig_TestInit {
+/// @title SystemConfig_Uncategorized_Test
+/// @notice General tests that are not testing any function directly of the `SystemConfig` contract
+///         are testing multiple functions at once.
+contract SystemConfig_Uncategorized_Test is SystemConfig_TestInit {
```
Shouldn't this be `SystemConfig_SuperchainConfig_Test`?

aliersh replied:
fixed in [7679be0](https://github.com/ethereum-optimism/optimism/pull/16561/commits/7679be02f0486417b4ea714491190cf349c44460)

aliersh committed:
fix(test): fix SystemConfig test contract naming

- Rename SystemConfig_Uncategorized_Test to SystemConfig_SuperchainConfig_Test to follow proper naming conventions

JosepBove commented:
LGTM; It would be nice to have a document somewhere explaining the changes and sharing it between the teams that work on contracts. (Probably in #smart-contracts in Discord)
